File: app/src/main/java/ar/com/wolox/android/example/ui/viewpager/request/RequestFragment.java
package ar.com.wolox.android.example.ui.viewpager.request;

import android.support.v7.widget.Toolbar;
import android.widget.TextView;

import ar.com.wolox.android.R;
import ar.com.wolox.wolmo.core.fragment.WolmoFragment;
import ar.com.wolox.wolmo.core.util.ToastFactory;

import javax.inject.Inject;

import butterknife.BindView;

public class RequestFragment extends WolmoFragment<RequestPresenter> implements IRequestView {

    @BindView(R.id.fragment_page2_toolbar) Toolbar mToolbar;
    @BindView(R.id.fragment_page2_title) TextView mTitle;
    @BindView(R.id.fragment_page2_body) TextView mBody;

    @Inject ToastFactory mToastFactory;

    @Inject
    public RequestFragment() {}

    @Override
    public int layout() {
        return R.layout.fragment_request;
    }

    @Override
    public void init() {
        mToolbar.setTitle(getString(R.string.page_2_toolbar_title));
    }

    @Override
    public void setNewsTitle(String title) {
        mTitle.setText(title);
    }

    @Override
    public void setNewsBody(String body) {
        mBody.setText(body);
    }

    @Override
    public void showError() {
        mToastFactory.show(R.string.unknown_error);
    }
}

File: app/src/main/java/ar/com/wolox/android/example/ui/viewpager/request/RequestFragment.kt
package ar.com.wolox.android.example.ui.viewpager.request

import ar.com.wolox.android.R
import ar.com.wolox.android.databinding.FragmentRequestBinding
import ar.com.wolox.wolmo.core.fragment.WolmoFragment
import ar.com.wolox.wolmo.core.util.ToastFactory
import javax.inject.Inject

class RequestFragment @Inject constructor() : WolmoFragment<FragmentRequestBinding, RequestPresenter>(), RequestView {

    @Inject internal lateinit var toastFactory: ToastFactory

    override fun layout() = R.layout.fragment_request

    override fun init() {
    }

    override fun setListeners() {
        binding.searchButton.setOnClickListener { presenter.onSearchRequested(binding.idInput.text.toString().toIntOrNull()) }
    }

    override fun setTitle(title: String) {
        binding.pageTitle.text = title
    }

    override fun setBody(body: String) {
        binding.pageBody.text = body
    }

    override fun showInvalidInput() = toastFactory.show(R.string.fragment_request_invalid_input)

    override fun showError() = toastFactory.show(R.string.unknown_error)
}
